### PR TITLE
List utf8.manifest as normal source file for MSVC

### PR DESCRIPTION
**Pull request by wantehchang**
List utf8.manifest as normal source file for MSVC

For toolchains targeting the MSVC ABI, CMake has first-class support for
embedding manifests. Manual embedding through a .rc source is needed
only when targeting the GNU ABI on Windows (MinGW).

Suggested by Brad King.

Fix https://github.com/AOMediaCodec/libavif/issues/2442.

### CODE REVIEW

**Review by jzern — APPROVED** [review]

**Comment by wantehchang** [comment]
@jzern James: I updated this pull request to the fix that Brad King suggested in https://github.com/AOMediaCodec/libavif/issues/2442#issuecomment-2377364754. Please review again. Thanks!

@bradking Brad: Please make sure I implemented your suggestion correctly. Thanks!

**Comment by bradking** [comment]
Yes, the diff as of commit 4362ac61487b19e62e79017396ca5c23f98c80fa is what I suggested, thanks.

**Comment by wantehchang** [comment]
@kmilos Miloš: I'd appreciate it if you could also take a look. Thanks!

@bradking My concern with the `if(MSVC) elseif(MINGW)` form is that it will silently ignore the utf8.manifest file if neither `MSVC` nor `MINGW` is true. I wonder if we should use the  `if(MSVC) else()` or  `if(MINGW) else()` form instead:

**Comment by bradking** [comment]
MSVC and MINGW are the only two ABI possibilities inside `if(WIN32)`, but if you prefer, `if(MSVC) .manifest else() .rc endif()` is probably best.

File: CMakeLists.txt [hunk 1]
```diff
@@ -718,17 +718,12 @@ endif()
...
-        # don't use the /MANIFEST:NO linker option:
-        #   /MANIFEST:EMBED,ID=1" failed (exit code 1123) with the following
-        #   output:
-        #   CVTRES : fatal error CVT1100: duplicate resource.  type:MANIFEST,
-        #   name:1, language:0x0409
-        if(NOT MINGW)
-            target_link_options(avifenc PRIVATE /MANIFEST:NO)
+        if(MSVC)
+            target_sources(avifenc PRIVATE apps/utf8.manifest)
+        elseif(MINGW)
```
Comment (jzern) [inline CMakeLists.txt:723]: I leave making this an `else()` as Brad suggested up to you.

**Review by jzern — APPROVED** [review]

**Pull request merged**